lib/gitlab/configuration.rb

```diff
@@ -32,6 +32,7 @@ def options
     def reset
       self.endpoint       = ENV['GITLAB_API_ENDPOINT']
       self.private_token  = ENV['GITLAB_API_PRIVATE_TOKEN']
+      self.auth_token     = ENV['GITLAB_API_AUTH_TOKEN']
```

Comment:
There is no case when user needs to set both `auth_token` and `private_token`. 
So, I propose to have only 1 config key (`private_token`), add alias (`oauth_token`) for it and auto detect correct header later behind the scenes. Does it make sense?